[functional] Use the saved device on storage instead for device_custom

anijain2305 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #163019
* __->__ #162987

Trying to reduce the number of __torch_dispatch__ calls of FakeTensorMode in the AOT metadata collection pass.


cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @kadeng @chauhang @amjames @Lucaskabela

anijain2305 committed:
[functional] Use the saved device on storage instead for device_custom

[ghstack-poisoned]

Lucaskabela reviewed (approved):
Thanks for the comment here to make sure the semantics are clear :)

anijain2305 committed:
Update on "[functional] Use the saved device on storage instead for device_custom"


Trying to reduce the number of __torch_dispatch__ calls of FakeTensorMode in the AOT metadata collection pass.


[ghstack-poisoned]

bdhirsh reviewed (approved):
nice catch!

anijain2305 committed:
Update on "[functional] Use the saved device on storage instead for device_custom"


Trying to reduce the number of __torch_dispatch__ calls of FakeTensorMode in the AOT metadata collection pass.


[ghstack-poisoned]
